src/utils/reductions.hpp

```diff
@@ -44,17 +46,21 @@ enum MPI_Op {
 #endif
 
 // Some helper functions
-template <typename U>
-void *GetPtr(std::vector<U> &v) {
+// Generic for containers with a data() method, including Kokkos::View and std::vector
+template <template <typename U, typename... Args> class Container, typename U,
+          typename... Args>
+void *GetPtr(Container<U, Args...> &v) {
   return v.data();
 }
 template <typename U>
 void *GetPtr(U &v) {
   return &v;
 }
 
-template <typename U>
-int GetSize(std::vector<U> &v) {
+// Generic for containers with a size() method, including Kokkos::View and std::vector
+template <template <typename U, typename... Args> class Container, typename U,
+          typename... Args>
+int GetSize(Container<U, Args...> &v) {
```

Comment:
This might be made more generic with 
```C++
template <class T> auto GetSizeImp(T& v, int) -> decltype(v.size()) {return v.size();}
template <class T> auto GetSizeImp(T& v, double) -> int {return 1;}
template <class T> auto GetSize(T& v) -> decltype(GetSizeImp(v, int())) { 
    return GetSizeImp(std::forward<T&>(v), int());
}
```
and something similar for `GetPtr` above

Reply:
Hmm that's interesting. I agree that would work and is more generic.

Reply:
I tried this briefly and ran into a few inconsistencies with other parts of the reduce infrastructure. So I think I'll skip this for now. But I'll create an issue.